### PR TITLE
Improve `$poll-export-status` to include all required attributes. 

### LINKED ISSUE #4051: Bulk Export Inconsistencies
Two issues need rectifying: 

1. After a bulk export, attempting to read the resulting binary while using an `Accept` header of `application/fhir+ndjson` should stream the data back to the client as NDJSON. Currently, it returns an XML binary resource. If the Accept header is omitted, currently the results are streamed back as JSON, not NDJSON. 

2. `$export-poll-status` is not currently sufficient. According to the HL7 [specification](http://hl7.org/fhir/uv/bulkdata/2021May/export.html#response---complete-status%7Chttp://hl7.org/fhir/uv/bulkdata/2021May/export.html%23response---complete-status%7Csmart-link) for bulk export,  the poll response for a completed Bulk Export job is required to include four elements:
       
      {{transactionTime}} 
      {{request}}
      {{requiresAccessToken}}
      {{output}}

Currently HAPI-FHIR only returns {{transactionTime}} and {{output}}.

### PR DESCRIPTION
**Pull request by tadgh**
Closes #4051

* Add all needed attributes to the bulk export polling response when it is completed. 
* Set `requiresAccessToken` to be true. 
* Set `request` to be the originating request URL which caused this bulk export to occur. This necessitated modifying the contents of the BulkExportJobParameters so it could be propagated all the way to the Report Writer step. 
* Add test.

### CODE REVIEW

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkExportUseCaseTest.java [hunk 1]
```diff
@@ -59,6 +63,42 @@ public class BulkExportUseCaseTest extends BaseResourceProviderR4Test {
...
+			myBatch2JobHelper.awaitJobCompletion(jobId);
+
+			//Then: When the poll shows as complete, all attributes should be filled.
+			HttpGet statusGet = new HttpGet(pollingLocation);
+			try (CloseableHttpResponse status = ourHttpClient.execute(statusGet)) {
+				String responseContent = IOUtils.toString(status.getEntity().getContent(), StandardCharsets.UTF_8);
+				ourLog.info(responseContent);
+
+				BulkExportResponseJson result = JsonUtil.deserialize(responseContent, BulkExportResponseJson.class);
+				assertThat(result.getRequest(), is(equalTo(myClient.getServerBase() + "/$export?_type=Patient")));
```
Comment (fil512) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkExportUseCaseTest.java:94]: nitpick: when a test asserts a response contains a string from the request, best practice is to put it in a variable

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java [hunk 2]
```diff
@@ -282,16 +287,18 @@ public void exportPollStatus(
 					BulkExportResponseJson bulkResponseDocument = new BulkExportResponseJson();
 					bulkResponseDocument.setTransactionTime(info.getEndTime()); // completed
 
+					bulkResponseDocument.setRequiresAccessToken(true);
```
Comment (fil512) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java:290]: I see now why you were asking about this.  If we're not enforcing this, how does it affect anything setting this to true?  Or is this just informative for audit purposes?  Anyway, I assume it's for compliance for now, but post-merge I would like to understand what this means.

**Review by fil512 — APPROVED** [review]

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkExportUseCaseTest.java (hunk 1 above)
Comment (tadgh) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkExportUseCaseTest.java:94]: Fixed, good call.

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java:290]: ok cool 👍🏻

**Pull request merged**